lib/checkstl.cpp

```diff
@@ -333,13 +333,13 @@ void CheckStl::mismatchingContainers()
                 std::map<const Variable *, unsigned int>::const_iterator it = containerNr.find(c);
                 if (it == containerNr.end()) {
                     for (it = containerNr.begin(); it != containerNr.end(); ++it) {
-                        if (it->second == i->container) {
+                        if (static_cast<int>(it->second) == i->container) {
```

Comment:
the compiler is so stupid! I assume that it warned about "unsigned expression == signed expression". 

This comparison will work as expected except in a very special case when a negative value is compared against a very large positive value and the bit pattern matches exactly. Like:  -1 == 0xFFFFFFFF.

Your cast does not solve any problem. If the comparison is -1 == 0xFFFFFFFF then the result will be the same with or without the cast.